docs/tutorial.md

```diff
@@ -3,217 +3,904 @@ id: tutorial
 title: Tutorial
 ---
 
-## Learn Formik by Building it.
+## Before we start
 
-My talk at React Alicante is really the best way to get started. It introduces the library (by watching me build a mini version of it) and demos how to build a non-trivial form (with arrays, custom inputs, etc.) using the real thing. It's around 54 minutes long.
+Welcome to the Formik tutorial. This will teach you everything you need to know to build simple and complex forms in React.
 
-<iframe width="800" height="315" src="https://www.youtube.com/embed/oiNtnehlaTo" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen title="Taming Forms in React - Jared Palmer"></iframe>
+If you're impatient and just want to start hacking on your machine locally, checkout [the 60-second quickstart](@todo).
 
-## Basics
+### What are we building?
 
-_If you just want to jump into some code, here's a quick walkthrough of the basics..._
+In this tutorial, we'll show how to build a complex newsletter signup form with React and Formik.
 
-Imagine you want to build a form that lets you edit user data. However, your
-user API has nested objects like so.
+You can see what we'll be building here: [Final Result](https://codesandbox.io/s/formik-v2-tutorial-final-t5fe0). If the code doesn't make sense to you, don't worry! The goal of this tutorial is to help you understand Formik.
```

Comment:
This line should have a bit more of a caveat that _the way the tutorial begins isn't really how we'd recommend someone starting out builds with Formik_. I think we get a lot of questions in the chat and with issues concerning how a format like the introductory code doesn't work with `useField`.

I think #1722 can help with that, but I also think this line can be more descriptive that the goal is to get to the end which is totally different.

"We're going to start by building a Formik form manually, which will eventually evolve into using the extremely simple and recommended API provided by Formik. The manual method can still be used for more complex scenarios etc etc, but is here mainly to demonstrate how the API works internally."

Reply:
Want to jump on discord or a zoom and collab on this?

Reply:
I had limited time this morning and spent it reading through this 😆 but I think calling this "Thinking in Formik" and creating a separate "Quickstart Tutorial" where this Thinking in Formik document basically builds the Quickstart example manually and demonstrates how the Quickstart example works underneath, then the Quickstart starts with something like "How is it this easy?! Learn how with "Thinking in Formik ->"" or something.

Reply:
I may have some async import tasks later where I can squeeze a quick chat in while I'm waiting, but I don't want to promise. I'll ping you in Discord if I do, and if it works out, it works out!

Reply:
Sounds good. I agree 100% with the split between quick start and thinking in formik